crates/api_common/src/request.rs

```diff
@@ -63,6 +91,16 @@ pub async fn fetch_link_metadata(url: &Url, context: &LemmyContext) -> LemmyResu
     .await?
     .error_for_status()?;
 
+  // Manually follow one redirect, using internal IP check
+  let location = response
+    .headers()
+    .get(LOCATION)
+    .and_then(|l| l.to_str().ok());
+  if let Some(location) = location {
+    let url = location.parse()?;
+    return Box::pin(fetch_link_metadata(&url, context)).await;
+  }
```

Comment:
This allows for infinite redirects and infinite recursion (so stack overflow). Need to rewrite. Same in the library PR.